src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/visitbalances/VisitBalanceResource.kt

```diff
@@ -29,19 +37,74 @@ class VisitBalanceResource(
     val activePrisonersCount = nomisApiService.getActivePrisoners(0, 1).totalElements
 
-    telemetryClient.trackEvent("visitbalance-reports-reconciliation-requested", mapOf("active-prisoners" to activePrisonersCount.toString()))
+    telemetryClient.trackEvent(
+      "visitbalance-reports-reconciliation-requested",
+      mapOf("active-prisoners" to activePrisonersCount.toString()),
+    )
     log.info("Visit Balance reconciliation report requested for $activePrisonersCount active prisoners")
 
     reportScope.launch {
       runCatching { reconciliationService.generateReconciliationReport(activePrisonersCount) }
         .onSuccess {
           log.info("Visit balance reconciliation report completed with ${it.size} mismatches")
-          telemetryClient.trackEvent("visitbalance-reports-reconciliation-report", mapOf("mismatch-count" to it.size.toString(), "success" to "true"))
+          telemetryClient.trackEvent(
+            "visitbalance-reports-reconciliation-report",
+            mapOf("mismatch-count" to it.size.toString(), "success" to "true"),
+          )
         }
         .onFailure {
           telemetryClient.trackEvent("visitbalance-reports-reconciliation-report", mapOf("success" to "false"))
           log.error("Visit balance reconciliation report failed", it)
         }
     }
   }
+
+  @PreAuthorize("hasRole('NOMIS_VISIT_BALANCE')")
+  @GetMapping("/visit-balance/reconciliation/{prisonNumber}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Run the reconciliation for this prison number",
+    description = """Retrieves the differences for a prisoner. Empty response returned if no differences found. 
+      Requires NOMIS_VISIT_BALANCE""",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Reconciliation differences returned",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint. Requires NOMIS_VISIT_BALANCE",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Offender does not exist",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  suspend fun generateReconciliationReportForPrisoner(
+    @Schema(description = "Prison number aka noms id / offender id display", example = "A1234BC")
+    @PathVariable prisonNumber: String,
```

Comment:
May want a different result (than match) if offender or visit balance doesn't exist in both Nomis and Dps - will do in separate PR